pkg/healthz/healthz_test.go

```diff
@@ -24,27 +27,41 @@ func TestHealthz(t *testing.T) {
 	tt := []struct {
 		path          string
 		kmsEncryptErr error
+		shouldSucceed bool
 	}{
 		{
 			path:          "/test-healthz-default",
 			kmsEncryptErr: nil,
+			shouldSucceed: true,
 		},
 		{
 			path:          "/test-healthz-fail",
 			kmsEncryptErr: errors.New("fail encrypt"),
+			shouldSucceed: false,
+		},
+
+		{
+			path:          "/test-healthz-fail-with-internal-error",
+			kmsEncryptErr: awserr.New(kms.ErrCodeInternalException, "test", errors.New("fail")),
+			shouldSucceed: false,
+		},
+		// user-induced errors should still fail "/healthz"
+		{
+			path:          "/test-healthz-fail-with-user-induced-invalid-key-state",
+			kmsEncryptErr: awserr.New(kms.ErrCodeInvalidStateException, "test", errors.New("fail")),
+			shouldSucceed: false,
+		},
+		{
+			path:          "/test-healthz-fail-with-user-induced-invalid-grant",
+			kmsEncryptErr: awserr.New(kms.ErrCodeInvalidGrantTokenException, "test", errors.New("fail")),
+			shouldSucceed: false,
 		},
 	}
 	for i, entry := range tt {
 		func() {
```

Comment:
No change required, but you could use [`testing.T.Run()`](https://pkg.go.dev/testing#T.Run) and add a `name` field to your test table cases
```suggestion
		t.Run(entry.name, func(t *testing.T) {
```